gateway/src/apicast/policy/jwt_claim_check/readme.md

```diff
@@ -0,0 +1,134 @@
+resource will be allowed without JWT constraint.
+
+In case of a GET request, the JWT needs to have the role claim as admin, if not
+the request will be denied. 
+
+## Examples
+
+- When you want to allow those who have the jwt claim role `role1` to access `/resource1`.
+
+```json
+{
+  "name": "apicast.policy.jwt_claim_check",
+  "configuration": {
+      "error_message": "Invalid JWT check",
+      "rules": [
+          {
+              "operations": [
+                  {"op": "==", "jwt_claim": "role", "jwt_claim_type": "plain", "value": "role1"}
+              ],
+              "combine_op":"and",
+              "methods": ["ANY"],
+              "resource": "/resource", 
+              "resource_type": "liquid"
+          }
+      ]
+  }
+}
+```
+
+- When you want to allow those who have the jwt claim role `role1` and `role2` to
+  access `/resource1` a OR combine operations can be used.
+
+```json
+{
+  "name": "apicast.policy.jwt_claim_check",
+  "configuration": {
+      "error_message": "Invalid JWT check",
+      "rules": [
+          {
+              "operations": [
+                  {"op": "==", "jwt_claim": "role", "jwt_claim_type": "plain", "value": "role1"},
+                  {"op": "==", "jwt_claim": "role", "jwt_claim_type": "plain", "value": "role2"}
+              ],
+              "combine_op":"or",
+              "methods": ["ANY"],
+              "resource": "/resource", 
+              "resource_type": "liquid"
+          }
+      ]
+  }
+}
+```
+
+- When you want to allow those who have the client role including the client ID
+  of the application client (the recipient of the access token) to access
+  `/resource1`. Use the `"liquid"` to specify the JWT information to the
+  `"name"` of the client role.
+
+```json
+{
+  "name": "apicast.policy.jwt_claim_check",
+  "configuration": {
+      "error_message": "Invalid JWT check",
+      "rules": [
+          {
+              "operations": [
+                  {"op": "==", "jwt_claim": "role_{{aud}}", "jwt_claim_type": "liquid", "value": "client1"},
+              ],
+              "combine_op":"and",
+              "methods": ["ANY"],
+              "resource": "/resource", 
+              "resource_type": "liquid"
+          }
+      ]
+  }
+}
+```
+
+- When you want to allow those who have who have the client `client1`'s role
+  `role1` to access the resource including the application client ID. Use the
+  `"liquid"` to specify the JWT information to the `"resource"`.
+
+```json
+{
+  "name": "apicast.policy.jwt_claim_check",
+  "configuration": {
+      "error_message": "Invalid JWT check",
+      "rules": [
+          {
+              "operations": [
+                  {"op": "==", "jwt_claim": "role", "jwt_claim_type": "plain", "value": "client1"},
+              ],
+              "combine_op": "and",
+              "methods": ["ANY"],
+              "resource": "/resource_{{jwt.aud}}", 
+              "resource_type": "liquid"
+          }
+      ]
+  }
+}
+```
```

Comment:
Would be good to show an example with more than one rule to show that it's supported.